### PR TITLE
Insecure rest

### PR DESCRIPTION
**Pull request by diegolovison**
Insecure rest

### CODE REVIEW

File: plugins/resteasy-http/src/main/java/org/radargun/http/service/RESTEasyCacheInfo.java [hunk 1]
```diff
@@ -74,11 +77,11 @@ public long getTotalSize() {
                size = Long.parseLong(responseStr);    //if the call was done to nodejs app, then the long size is returned
 
             } catch (NumberFormatException ex) {
-               //If the call was done to Infinispan Server, then the response is the list of all entries
-               //WARNING: As this way the REST API retrieves all keys from the cache and returns them as a response,
-               //this part of block may have an impact on the performance of the stage which rely on this trait.
-               String[] entries = responseStr.split(System.getProperty("line.separator"));
-               size = entries.length;
+               Matcher matcher = Pattern.compile("\"size\":(\\d+)").matcher(responseStr);
```
Comment (gustavolira) [inline plugins/resteasy-http/src/main/java/org/radargun/http/service/RESTEasyCacheInfo.java:80]: Nitpick: It's more readable and easier to understand if you create a private method with a name which explain what you are doing ou desire here, it's more readable than analyse the pattern, find, parse ...

File: plugins/resteasy-http/src/main/java/org/radargun/http/service/AbstractRESTEasyService.java [hunk 2]
```diff
@@ -61,6 +63,12 @@
    @Property(doc = "Http headers for request. Default is null", complexConverter = KeyValueProperty.KeyValuePairListConverter.class)
    private List<KeyValueProperty> httpHeaders;
 
+   @Property(doc = "If the REST client will ignore the https certificates")
```
Comment (Crumby) [inline plugins/resteasy-http/src/main/java/org/radargun/http/service/AbstractRESTEasyService.java:66]: Ignoring https certificates and using insecure http protocol are two different things. From my understanding insecure property does the latter while for ignoring https certificates there is `trustAll` bellow.

Comment (diegolovison) [reply plugins/resteasy-http/src/main/java/org/radargun/http/service/AbstractRESTEasyService.java:66]: Maybe change to:
* protocol=(http|https)
* trust-all=(true|false)

WDYT?

Comment (Crumby) [reply plugins/resteasy-http/src/main/java/org/radargun/http/service/AbstractRESTEasyService.java:66]: Sounds good

File: plugins/resteasy-http/src/main/java/org/radargun/http/service/AbstractRESTEasyService.java [hunk 3]
```diff
@@ -73,6 +81,10 @@ public synchronized void start() {
          return;
       }
 
+      if (trustAll) {
+         HttpsHelper.trustAll();
```
Comment (Crumby) [inline plugins/resteasy-http/src/main/java/org/radargun/http/service/AbstractRESTEasyService.java:85]: Setting trustall globally as default if one of the clients sets it isn't a good design. But probably shouldn't be issue for RadarGun as you are probably not going to run performance tests against cluster with valid and invalid certificats.

File: plugins/resteasy-http/src/main/java/org/radargun/http/service/RESTEasyCacheInfo.java [hunk 4]
```diff
@@ -74,11 +77,11 @@ public long getTotalSize() {
...
+               if (matcher.find()) {
+                  // size:100
```
Comment (Crumby) [inline plugins/resteasy-http/src/main/java/org/radargun/http/service/RESTEasyCacheInfo.java:82]: Is this comment an example of what size can be returned or is 100 magic number that isn't clear by default?

File: plugins/resteasy-http/src/main/java/org/radargun/http/service/AbstractRESTEasyService.java (hunk 3 above)
Comment (diegolovison) [reply plugins/resteasy-http/src/main/java/org/radargun/http/service/AbstractRESTEasyService.java:85]: It is false by default

File: plugins/resteasy-http/src/main/java/org/radargun/http/service/RESTEasyCacheInfo.java (hunk 4 above)
Comment (diegolovison) [reply plugins/resteasy-http/src/main/java/org/radargun/http/service/RESTEasyCacheInfo.java:82]: Removed the comment

**Review by Crumby — APPROVED** [review]

**Comment by gustavolira** [comment]
Merged #676 into master. Thanks Diego

File: plugins/resteasy-http/src/main/java/org/radargun/http/service/AbstractRESTEasyService.java (hunk 3 above)
Comment (Crumby) [reply plugins/resteasy-http/src/main/java/org/radargun/http/service/AbstractRESTEasyService.java:85]: I didn't mean the property itself but the fact that one client setting the property to true sets static variables in `HttpsURLConnection`:
```
HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
HttpsURLConnection.setDefaultHostnameVerifier(validHosts);
```

**Pull request merged**